libraries/fc/include/fc/log/trace.hpp

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+#ifdef ENABLE_CPPKIN
+
+#include <fc/scoped_exit.hpp>
+#include <cppkin.h>
+
+/// Simple wrappers for cppkin tracing
+
+inline bool fc_cppkin_trace_enabled = false;
+
+/// @param VARNAME unique within scope variable name for created trace
+/// @param TRACE_STR const char* identifier for trace
+#define fc_create_trace( VARNAME, TRACE_STR ) \
+  std::optional<cppkin::Trace> VARNAME; \
```

Comment:
In macros, it's a generally good idea to be extra paranoid and use `::std`, `::cppkin`, `::&c.`

Reply:
ok